### ISSUE TITLE
User on Windows with group incorrectly warns about passing GID

### ISSUE THREAD

**Issue by tas50** [chef/chef#5521]
## Description

If you use the user resource and specify a group for the user the group is passes as GID and you receive a warning message saying you should use group instead (which you did).

## Chef Version

12.15

## Platform Version

Windows 2012R2

## Replication Case

```ruby
user 'foobarbaz' do
  group 'foobarbaz'
end
```

## Client Output

```
         * group[foobarbaz] action create
           - create group foobarbaz
         * windows_user[foobarbaz] action create[2016-11-07T13:01:24-08:00] WARN: The 'gid' attribute is not implemented by the Windows platform. Please use the 'group' resource to assign a user to a group.

more stuff...
```

**Comment by btm** [comment]
Should the user resource create missing groups? This could be a separate issue.

Currently this resource is not creating the missing group. Either it should raise an error, or create the missing group.

**Comment by smurawski** [comment]
We probably are not erroring correctly here, since we'd never be able to add that user to that group.

We have an added challenge with this particular case, as we have the user name and group name being the same, which you can't have on Windows.

See https://github.com/chef/chef/issues/5857#issuecomment-288106226 for more comments on the same user/group names.

**Comment by harikesh-kolekar** [comment]
group or gid` attribute is not supported for the `user` resource on windows platform. Even if `group` attribute is passed in `user` resource, it gets ignored.
The warning msg `The 'gid' attribute is not implemented by the Windows platform. Please use the 'group' resource to assign a user to a group.` is correct. It suggests to use the `group resource`(not `group` attribute) to add user to the group. 
@btm, should we leave this warning as it is or raise an error if user is passing `group or gid` attribute for windows?

**Comment by smurawski** [comment]
@harikesh-kolekar This should remain a warning, but be clarified.

Problems with the existing warning:
- Only mentions `gid` when the user supplied `group`.  We should add a mention of the `group` attribute.
- Refers to the `group` resource, but does not mention how to specify group membership.

The original report from @tas50 notes that he was using the `group` resource to create the group already.  As an experienced Chef user, if he is being mislead by the warning, we need to clarify.

An example of what I think the updated warning should look like is:

```
The 'gid' (or 'group') property is not implemented on the Windows platform. Please use the `members` property of the  'group' resource to assign a user to a group.
```

**Comment by smurawski** [comment]
Fixed by #6125
